### PR TITLE
feat: add the README file

### LINKED ISSUE #1: docs: Initialize README.md file.
## Description
As a contributor,
I need to learn about the project,
so that I can collaborate with others on its development.

## Mocks
```
# bridge-in-tech-android


## Contributing

**This project is under active development**

Please read our [Contributing Guidelines](https://github.com/anitab-org/bridge-in-tech-backend/blob/develop/.github/CONTRIBUTING.md), [Code of Conduct](https://github.com/anitab-org/bridge-in-tech-backend/blob/develop/.github/CODE_OF_CONDUCT.md) and [Reporting Guidelines](https://github.com/anitab-org/bridge-in-tech-backend/blob/develop/.github/REPORTING_GUIDELINES.md) thoroughly.

## Branches

- **master**: This branches contain the deployment of the backend.
- **develop**: This contains the latest code. All the contributing PRs must be sent to this branch.

## Contact

If you have any questions or want to discuss something about this repo, feel free to join our [Zulip Community](http://anitab-org.zulipchat.com/). If you are a new contributor, head over to this project's stream [#bridge-in-tech](https://anitab-org.zulipchat.com/#narrow/stream/237630-bridge-in-tech) on Zulip to see ongoing discussions.

## License

The project is licensed under the GNU General Public License v3.0. Learn more about it in the [LICENSE](https://github.com/anitab-org/anitab-org.github.io/blob/develop/LICENSE) file.
```

### Update [Required]
- [ ] Include the minimum details that were included in the **Mocks** section above.
### Enhancement to Update [Optional]
- [ ] Reach out to @mtreacy002 for a description of the project!

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor.

## Estimation
0.5 hours

### PR DESCRIPTION
**Pull request by menina763**
### Description

Added README file to bridge-in-tech repo

Fixes #1

### Type of Change:

- Documentation

### Checklist:

- [X] My PR follows the style guidelines of this project
- [X] I have performed a self-review of my own code or materials
- [X] I have made corresponding changes to the documentation
- [X] Any dependent changes have been merged

### CODE REVIEW

File: .github/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Bridge-In-Tech (android)
```
Comment (mtreacy002) [inline .github/README.md]: > (android)

This repo is actually the 'backend' of the Bridge-In-Tech system, @Menina294. The 'android' is the frontend, and it's another [repo](https://github.com/anitab-org/bridge-in-tech-android) of bridge-in-tech. Can you please fix this 😉 ?

File: .github/README.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+
+## Contributing
+
+**This project is under active development**
+
+Bridge-In-Tech is an extension of Mentorship System that allows industry companies, mentors and students to actively contribute to one another.
```
Comment (mtreacy002) [inline .github/README.md]: > industry companies

can we put '/' in between these, coz they are actually similar. Also, make sure use all in 'plural' forms 😉 .
> contribute to

can we use 'collaborate with' instead?

File: .github/README.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# Bridge-In-Tech (android)
+
+
+## Contributing
+
+**This project is under active development**
+
+Bridge-In-Tech is an extension of Mentorship System that allows industry companies, mentors and students to actively contribute to one another. 
+
+This is the Android client of Bridge-In-Tech.
```
Comment (mtreacy002) [inline .github/README.md]: > Android 

please do the same as previous comment 😃

**Review by mtreacy002 — CHANGES_REQUESTED** [review]
Great start, @Menina294. Well done on putting the contributing/reporting guidelines, code of conduct and license information on this README.md as well 👍 (I'm wondering what else to put in here but never even thought about them 😊 ).
I do think you need to make some small changes though. Let me know if you have different opinion. 😉.

File: .github/README.md (hunk 1 above)
Comment (isabelcosta) [reply .github/README.md]: +1

File: .github/README.md (hunk 2 above)
Comment (isabelcosta) [reply .github/README.md]: I also wouldn't say it is an extension of the mentorship system, the purpose is different than mentorship system. Right?
Also, this description should be directly below the title of the README, and not under Contributing header. Makes sense?

File: .github/README.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+Please read our [Contributing Guidelines](https://github.com/anitab-org/bridge-in-tech-backend/blob/develop/.github/CONTRIBUTING.md), [Code of Conduct](https://github.com/anitab-org/bridge-in-tech-backend/blob/develop/.github/CODE_OF_CONDUCT.md) and [Reporting Guidelines](https://github.com/anitab-org/bridge-in-tech-backend/blob/develop/.github/REPORTING_GUIDELINES.md) thoroughly.
```
Comment (isabelcosta) [inline .github/README.md]: this makes sense coming right after "Contributing" section

Comment (isabelcosta) [reply .github/README.md]: Also could you use relative paths for these links, such as:

```suggestion
Please read our [Contributing Guidelines](.github/CONTRIBUTING.md), [Code of Conduct](.github/CODE_OF_CONDUCT.md) and [Reporting Guidelines](.github/REPORTING_GUIDELINES.md) thoroughly.
```

File: .github/README.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+- **master**: This branch contains the deployment of the backend.
+- **develop**: This contains the latest code. All the contributing PRs must be sent to this branch.
+
+## Contact
+
+If you have any questions or want to discuss something about this repo, feel free to join our [Zulip Community](http://anitab-org.zulipchat.com/)! If you are a new contributor, head over to this project's stream [#bridge-in-tech](https://anitab-org.zulipchat.com/#narrow/stream/237630-bridge-in-tech) on Zulip to see ongoing discussions.
+
+## License
+
+The project is licensed under the GNU General Public License v3.0. Learn more about it in the [LICENSE](https://github.com/anitab-org/anitab-org.github.io/blob/develop/LICENSE) file.
```
Comment (isabelcosta) [inline .github/README.md]:
```suggestion
The project is licensed under the GNU General Public License v3.0. Learn more about it in the [LICENSE](LICENSE) file.
```

Relative path, so this always links to the correct file. For example on your fork, GitHub will be able to redirect to the file on your fork repo :)

**Review by isabelcosta — CHANGES_REQUESTED** [review]
I requested some additional changes, but you are almost there @Menina294 ;)
Great contribution with the first version of the README 👏

File: .github/README.md (hunk 5 above)
Comment (mtreacy002) [reply .github/README.md]: > I also wouldn't say it is an extension of the mentorship system, the purpose is different than mentorship system. Right?

My bad, I think I'm to blame for this word 'extension' coz I used this in my wiki home page which is why probably @Menina294 use it here. Sorry again. I'll fix the wiki home page now 😆

**Comment by mtreacy002** [comment]
@Menina294, do you know how to make the changes requested on this PR? If you need help, just let me know 😉

**Comment by menina763** [comment]
Sorry I'm responding much later! And I was going to ask as to if I should try to resolve one change first, then the next etc. or if I can make all the requested changes at once?

**Comment by mtreacy002** [comment]
That's ok, @Menina294. Take your time. we're only going to check on you if no comment are made within 3 business days after the last one. You've just pushed the PR today, so, you're good 👍. 

> Sorry I'm responding much later! And I was going to ask as to if I should try to resolve one change first, then the next etc. or if I can make all the requested changes at once?

On this, you can make the changes all at once in your local IDE and push again in one go. Also, the best practice (which I just recently learn 😀 ) is to have one commit per PR (using squash commit command before you make the second push). But for now, I think it's ok for you to send the new changes just as a separate commit. The maintainers of this repo can squash merge it when this PR gets approved. @isabelcosta, can you confirm on this, please?

File: .github/README.md (hunk 5 above)
Comment (isabelcosta) [reply .github/README.md]: No problem, I actually think that your idea is one of its own. Maybe I am seeing this from a wrong angle @mtreacy002 if you say so, I don't mind :)

**Comment by isabelcosta** [comment]
@mtreacy002 is correct :) @Menina294 you can do as many commits as you would like, the frequency of commits is something that may depend from developer to developer. We will squash and merge in the end when it's all approved, and your commit history will remain intact in this PR. Let me know if there is anything that isn't clear @Menina294

**Comment by menina763** [comment]
Okay, I think I've made the neseccary changes to it! I just commited and pushed to my own repo, so what am I to do next? I'm not exactly sure 😅.

**Comment by mtreacy002** [comment]
Well done, @Menina294. Now, before we continue, I'd like to ask, what were the steps you've done when you fork the repo? I'm asking coz I want to know if there's some missing steps we need to add so you can successfully push to the AnitaB repo.

**Comment by menina763** [comment]
I think all I did was click the fork button on the "Bridge-In-Tech" repo.

**Comment by mtreacy002** [comment]
So, did you fork (clicking the top right fork button on the repo), or fork (same as before) and clone (running `git clone` command on your terminal?

**Comment by menina763** [comment]
I remember running it in VSCode but I don't think I did in my terminal.
Also what's a terminal?

**Comment by mtreacy002** [comment]
In VSCode, the terminal is where you run the command line. If you run clone inside VSCode, then it means you run it in the terminal. 😉 

Screen Shot 2020-05-09 at 11 25 29 am

**Comment by mtreacy002** [comment]
Can you now run the following command from the terminal

> git remote -v

We want to check if you have setup the remote for your fork repo. If you have, you should see something like this, with exception of the name of repo (for your own: `your_name/bridge-in-tech-backend` for remote:`anitab-org/bridge-in-tech-backend`)

Screen Shot 2020-05-09 at 11 29 04 am

**Comment by menina763** [comment]
I got the following response:
`origin  https://github.com/Menina294/bridge-in-tech-backend (fetch)
origin  https://github.com/Menina294/bridge-in-tech-backend (push)
PS C:\Users\diana\bridge-in-tech-backend>`

**Comment by mtreacy002** [comment]
Ok, so that means you haven't got the upstream setup. That's ok, we can set it soon. Before we do, I noticed that you've made your changes on your `develop` branch. This is probably done before I showed you the [firstcontributions](https://github.com/firstcontributions/first-contributions) in the previous post. As suggested there (and hopefully you have followed their tutorial as it's important for beginner contributor to learn the git workflow), you need to create and checkout to a new branch and make your changes on that branch (so for future work, please don't make your changes on the develop branch 😉). But since you haven't been informed before, this mistakes is understandable, so don't feel bad 😃 .

For now, we'll try to cancel your push. So, you said you have commit and push to your origin. Or did you only commit but have not yet push?
Update: don't worry, I can see the new commit you've made on your fork repo. That's fine, we can just fo to the next step.

**Comment by mtreacy002** [comment]
Now, can you set the remote first before we checkout to the new branch? Let me know if you're not sure how to do this (hint: the tips on [mentorhip backend wiki]( https://github.com/anitab-org/mentorship-backend/wiki/Fork,-Clone-&-Remote#clone) is a good start 😉) - I just notice this tips is missing on firstcontributions site.
If you comfortable, set up the remote (remember to change the repository name to the one shown in the previous post)

**Comment by menina763** [comment]
So just to double check, I would run the following:
`git remote add upstream https://github.com/systers/bridge-in-tech-backend`
Correct?

**Comment by mtreacy002** [comment]
instead of Systers, can you change it to `anitab-org/bridge-in-tech-backend` as per the repo name you're currently working on 😉? The rest is good.

**Comment by menina763** [comment]
Ohh, okay!

**Comment by mtreacy002** [comment]
once you're done, run the same command [here](https://github.com/anitab-org/bridge-in-tech-backend/pull/2#issuecomment-626083679) and you should see the same thing (pls post a screenshot to confirm)

**Comment by menina763** [comment]
Alrighty, I think I got it now! And here's the screenshot as well.

image

**Comment by mtreacy002** [comment]
great. now you can create and checkout to a new branch. Are you comfortable with this? My suggestion for the branch name, keep it short, and add issue number at the front of branch name.

It'll be to make it easier for you to know which branch for what purpose later on

**Comment by menina763** [comment]
I don't think I've ever done it before so no 😅.

**Comment by mtreacy002** [comment]
for, example, I have this as one of my working branch

Screen Shot 2020-05-09 at 12 15 42 pm

**Comment by mtreacy002** [comment]
> I don't think I've ever done it before so no 😅.

that's ok, you probably don't remember, but you should have done it when you were doing the firstcontributions tutorial.

Screen Shot 2020-05-09 at 12 17 02 pm
 

But anyway, run the command on the screenshot above just replace the one inside `<>` with your own branch name. Note: another benefit putting issue number at the beginning (as shown in my example above) will also handy in situation where you are working on few different PR at the same time 😉

**Comment by menina763** [comment]
Hmm, I seem to get an error. 

image

Could I have forgotten something?

**Comment by mtreacy002** [comment]
sorry, I forgot to tell you. you should remove the `<>` brackets... 😊 . Just replace with only name without brackets 😉

**Comment by menina763** [comment]
Ohhh okay!

**Comment by menina763** [comment]
Okay, it says that I sucessfully switched to a new branch!

**Comment by mtreacy002** [comment]
Great!!. Now, on the file that you're working on (README.md), make small insignificant change just so VSCode picks it up as a new change (sometimes just adding a whitespace anywhere will do the trick 😂). Then do another commit. but do not push.

**Comment by menina763** [comment]
Ahh okay!

**Comment by menina763** [comment]
Okay, just finished commiting it!

**Comment by mtreacy002** [comment]
Sorry, I was thinking whether to show you the long way or the short way..

Maybe we'll do the short one for now. You can push it to the origin branch for now. (So first option on the screenshot)

Screen Shot 2020-05-09 at 12 45 33 pm

**Comment by menina763** [comment]
Okay, just finished pushing it!

**Comment by mtreacy002** [comment]
I forgot to tell you the [preferred commit message style](https://github.com/anitab-org/mentorship-backend/wiki/Commit-Message-Style-Guide) we use in AnitaB. But that's ok, you can follow this for next time.

**Comment by mtreacy002** [comment]
> Okay, just finished pushing it!

Now, you can go to your Github GUI (on browser - your forked repo) and you should see a yellow highlighted notification that you've just pushed a commit. You can click the Compare button on that highlighted notification.

**Comment by menina763** [comment]
So do I click "Compare and Pull Request"?

**Comment by mtreacy002** [comment]
Make sure you comparing it to your issue branch, not develop branch

Screen Shot 2020-05-09 at 12 54 12 pm

**Comment by mtreacy002** [comment]
When you click `create pull request`, you should see the PR template like this in the body of your PR

Screen Shot 2020-05-09 at 12 55 36 pm

change it accordingly. Don't forget to name the title as per the purpose of the PR. It will be similar to the commit style message as per AnitaB guidelines above

Screen Shot 2020-05-09 at 12 57 16 pm

**Comment by menina763** [comment]
So just to double check (sorry if I'm making a lot of double checks!), I would compare it like this:
image

And then once I make the necessary changes to the PR template, I submit a new PR correct? Also would I name the PR as `docs: add suggestions to README`?

**Comment by mtreacy002** [comment]
No wait, you have AnitaB there as master. pls change to develop branch of anitab

**Comment by menina763** [comment]
Ohh, my bad! I mixed up the branches. Okay, I'll make sure to compare it to the dev. branch 👍.

**Comment by menina763** [comment]
And then I make a new PR? I just want to make sure because I originally thought that I would keep working in this PR.

**Comment by mtreacy002** [comment]
> And then I make a new PR? I just want to make sure because I originally thought that I would keep working in this PR.

the thing is, coz when you made this PR, you made the change on the develop branch (which is NOT aligned with our workflow). The problem with this, although we could just accept it and merge to anitab develop branch, every now and then you'll have trouble coz it could cause merge conflict.  So, since you're new, I would like to show you the preferred workflow. 

It's up to you on which one you'd like to keep. we can do either way. The one not chosen will be closed, but if you choose to use the new PR, just mention it next to `Fixes #1`. So, do, `Fixes #1 and replaced PR#2`

Updated

**Comment by menina763** [comment]
Hmm, okay I think I would want to see the perferred workflow, even if it means I might have some issues.

**Comment by mtreacy002** [comment]
> Hmm, okay I think I would want to see the perferred workflow, even if it means I might have some issues.

so, you're keeping the new PR, right? don't forget to add that `replaced PR#2` note. sorry, I've updated my post before this. 😂

**Comment by menina763** [comment]
If by new PR you mean the one I was about to make right now then yes. 

If I may ask, what will happen to this PR?

**Comment by menina763** [comment]
And I'll make sure to add those notes as well!

**Comment by mtreacy002** [comment]
> If by new PR you mean the one I was about to make right now then yes.

That's right.
> If I may ask, what will happen to this PR?

This PR will be closed without merge, and then we will go to VSCode IDE to fix your develop branch so it's synced with the anitab develop branch.

**Comment by menina763** [comment]
Just made the PR! Now I don't think I made any mistakes, but please double check for me.

**Comment by mtreacy002** [comment]
> Just made the PR! Now I don't think I made any mistakes, but please double check for me.

Great. Will give my review soon 😉 .

Now, shall we go back to fix your develop branch?

**Comment by menina763** [comment]
Sure! So, what would be the first thing to do?

**Comment by mtreacy002** [comment]
Let's checkout back to your develop branch on the VSCode IDE

**Comment by menina763** [comment]
How do I make sure I'm on the right branch? Do I need to type something in the terminal?

**Comment by mtreacy002** [comment]
> How do I make sure I'm on the right branch? Do I need to type something in the terminal?

You can, alternatively just use the VSCode GUI for that

Click on the `source control' icon on the left navbar on VSCode, then you'll click `checkout to...` then select `develop`

Screen Shot 2020-05-09 at 1 52 06 pm

**Comment by menina763** [comment]
Ahh, okay! Now I'm on the develop branch.

**Comment by mtreacy002** [comment]
When you check the commits made on develop branch, you normally want to see that you're aligned with the upstream anitab develop branch (which can be done by routine fetching, especially before you create a new branch for working on an issue). 
Here, you can see your 1 commit ahead 

Screen Shot 2020-05-09 at 1 50 22 pm

When you go to `commits` on the top left you can see all the commits you've made including the old PR commit. 

Screen Shot 2020-05-09 at 1 47 09 pm

Note down the commit id of the one before that coz you want to put everything back as if the last commit didn't happened.

**Comment by mtreacy002** [comment]
So it should be `ba92976`

**Comment by mtreacy002** [comment]
then on your terminal run the following

> git reset --hard ba92976

this should put back your develop branch to the point before last commit was made, but for now it's only on your local git (Be careful there's a space between --hard and number. it's a bit too close on the image). you want to make sure your remote git also will reflect this. to do this, run. 

> git push -f origin develop

**Comment by mtreacy002** [comment]
After you completed the steps above, check your fork repo on develop branch by refreshing the browser. You should see that you are no longer ahead from the upstream anitab develop. Pls let me know how you go.

**Comment by menina763** [comment]
So once I run `git reset`, I then run `git push`?

Also, does this mean I ran it correctly?
image

**Comment by mtreacy002** [comment]
> So once I run `git reset`, I then run `git push`?
> 
> Also, does this mean I ran it correctly?
> image

It shouldn't be like that. 🤔. Can you show the screenshot of your bottom left corner of your VSCode pls

Screen Shot 2020-05-09 at 2 26 42 pm

**Comment by menina763** [comment]
Alrighty:

image

**Comment by mtreacy002** [comment]
ok, can you run and screenshot the output of

> git log

**Comment by menina763** [comment]
Okay:

image

There is still a lot more to the log. Do you want me to include that as well?

**Comment by mtreacy002** [comment]
No, that's ok, let me check something. I'll be back

**Comment by menina763** [comment]
Alright, I think I'll sign off for now since it's pretty late for me and I need wake up early for the design session. Hopefully we'll work on this after the session!

**Comment by mtreacy002** [comment]
You're

> Alright, I think I'll sign off for now since it's pretty late for me and I need wake up early for the design session. Hopefully we'll work on this after the session!

Ok, meanwhile I'll see how to solve the issue. more likely coz the command use in Mac (mine) and windows  (yours) are different. will let you know onceI find out something

**Comment by mtreacy002** [comment]
found it. you could try 

> git log --oneline --graph

then screenshot the result. After that, depends on the result, you'll do 
> git reset --hard HEAD~X

Where X is the number of commits you need to go back to the one we want (ba92976). 
But if you want, we can do it together like today, just let me know when you are ready.
Let me know if this works for you. More on this can be found [here](https://devconnected.com/how-to-git-reset-to-head/)

**Comment by isabelcosta** [comment]
> Just made the PR! Now I don't think I made any mistakes, but please double check for me.

@Menina294 great! I love seeing that @mtreacy002 help you on this and I am sure you learned a lot just with this help :) if you prefer, we can continue the work in the other PR :)

@Menina294
Can you try this on your `develop` branch locally.
```
git pull -f upstream develop
```
and then:
```
git push -f origin develop
```
and then, check the local commit history on `develop` branch:
```
git log --oneline
```
let us know what you see, because if these commands work like I think, what you should see is just 2 commits that are in this original repository.

**Comment by menina763** [comment]
Okay, so just to make sure, should I follow what Maya said first and then do what Isabel said?

**Comment by mtreacy002** [comment]
> Okay, so just to make sure, should I follow what Maya said first and then do what Isabel said?

You can do it either way, we could use my suggestion first if you want... What do you think? If you are going with mine, pls do a step at a time (show me screenshot of what you have after first step) 😉

**Comment by menina763** [comment]
I think I'll follow your suggestion, Give me a minute while I setup 😊.

Just to make sure, we were on the develop branch correct?

**Comment by mtreacy002** [comment]
correct

**Comment by menina763** [comment]
Here's what I got after typing `git log --oneline --graph`

image

**Comment by mtreacy002** [comment]
Ok, so it looks like you only need to move HEAD 2 commits back (coz we want to put it back to commit id ba92967).

Now you can try run 
> git reset --hard HEAD~2

**Comment by menina763** [comment]
Alright, here's the result

image

**Comment by mtreacy002** [comment]
Perfect. Your local branch now should be the same with AnitaB develop branch.

Now we just need to push this to your remote fork repo develop branch.

Do the following:

> git push -f origin develop

**Comment by mtreacy002** [comment]
If this command is successful, you should get this PR automatically closed and refresh your develop branch, you should see you're now align with Anita B develop repo. We can continue the discussion in your PR#3

**Pull request closed without merging**